C

GeoffreyDlvl opened the pull request.

GeoffreyDlvl committed:
Create .travis.yml

rparpa commented on .travis.yml:
```diff
@@ -0,0 +1,11 @@
+php:
+  - 5.5
+  - 5.6
+  - 7.4.1
```
Pas besoin d'aller aussi bas en versions de PHP.

Juste 7.4 c'est OK

GeoffreyDlvl committed:
Update .travis.yml

fsaintier committed:
Initialisation projet en JS

fsaintier committed:
Merge branch 'master' of https://github.com/GeoffreyDlvl/Trollop-Project

fsaintier committed:
Suppression PHP

fsaintier committed:
Travis conf

fsaintier committed:
Travis

fsaintier committed:
Initialisation fichier de tests postman

fsaintier committed:
Suppression repertoire back

fsaintier committed:
Je vais y arriver

fsaintier committed:
Ajout script start.ci dans package.json

fsaintier committed:
Ajout du fichier process.json

fsaintier committed:
Derniere tentative avant pendaison

rparpa commented on public/stylesheets/style.css:
```diff
@@ -0,0 +1,8 @@
+body {
```
Vous pouvez utiliser bootstrap.css (présent dans le skeleton original).

rparpa replied:
cc @Remi-Guijarro

fsaintier committed:
Modifications pour que l'appli se lance a nouveau

fsaintier committed:
Correction package.json

fsaintier committed:
Ajout www

fsaintier committed:
Ajout du lancement de node a l'install

fsaintier committed:
Correctif supplementaire

Commit:
Set up new project architecture

Commit:
Fixed : docker-compose and makefile to be able to build all the apps

Commit:
replaced pm2 by nodemon

fsaintier committed:
Modification de l'action start pour le back + ajout quelques détails pour le lancement dans le README

fsaintier committed:
Test Travis

Commit:
Merge pull request #10 from GeoffreyDlvl/refact-archi

Refact archi

fsaintier committed:
Allegement des Dockerfile

Remi-Guijarro committed:
Merge pull request #11 from GeoffreyDlvl/refact-archi

Allegement des Dockerfile

Commit:
Merge branch 'refact-archi' of https://github.com/GeoffreyDlvl/ParkTonChar

fsaintier committed:
Initialisation affichage de la carte un peu destructurée pour le moment

fsaintier committed:
Ajout du css leaflet pour restructuration de la carte

AdelinoAraujo committed:
Delete useless directories / files

AdelinoAraujo committed:
Update .gitignore

AdelinoAraujo committed:
Create empty readme.md for back-end

AdelinoAraujo committed:
Refactor back-end architecture

AdelinoAraujo committed:
Update makefile

AdelinoAraujo committed:
Add public folder in back-end

AdelinoAraujo committed:
fix back/public/index.html file after map testing

AdelinoAraujo committed:
Fix front/public/index.html after little, very little mistake

AdelinoAraujo committed:
Add make install command - testing

AdelinoAraujo committed:
Add api request tests

AdelinoAraujo committed:
Add user entity

AdelinoAraujo committed:
Add user entity back-end

AdelinoAraujo committed:
Update travis.yml

AdelinoAraujo committed:
Update travis.yml

Commit:
Merge pull request #12 from GeoffreyDlvl/travis-integration

Travis integration

Commit:
Merge branch 'master' into api-open-data-collection-test

Commit:
Merge pull request #14 from GeoffreyDlvl/api-open-data-collection-test

Add api request tests

Commit:
Merge branch 'master' into entity-user

Commit:
Merge pull request #15 from GeoffreyDlvl/entity-user

Entity user

Commit:
Merge branch 'master' of https://github.com/GeoffreyDlvl/ParkTonChar

Commit:
Add BootstrapVue support to project

Commit:
Implement basic header

Commit:
Implement basic layout for main page

Components will evolve and be replaced with actual components.

GeoffreyDlvl committed:
Merge pull request #16 from GeoffreyDlvl/baseLayout

Base layout

Commit:
Set up map component with MapBox :  this is a basic implementation

Commit:
Deleted un used leaflet css from main.js

AdelinoAraujo committed:
Add vehicule.js and vehicule.test.js to back-end

AdelinoAraujo committed:
Add vehicule.js and vehicule.test.js to front-end

AdelinoAraujo committed:
Add parkingSpot to back-end

AdelinoAraujo committed:
Add parkingSpot to front-end

AdelinoAraujo committed:
Update parking spot entity with vehiculeId

AdelinoAraujo committed:
Update entity-user with vehiculeId

Commit:
Merge pull request #17 from GeoffreyDlvl/entity-user

Update entity-user with vehiculeId

Commit:
Merge pull request #18 from GeoffreyDlvl/entity-vehicule

Entity vehicule

Commit:
Merge pull request #19 from GeoffreyDlvl/parkingSpot

Parking spot

Commit:
Add database schema image to rsc folder in project root

Commit:
Update db.sql

GeoffreyDlvl committed:
Merge pull request #20 from GeoffreyDlvl/databaseArchitecture

Database architecture

Commit:
Added mountbank

Commit:
Added Makefile target to update dependencies on docker

Commit:
Implemented mountbank basic stub

Commit:
Changed basic stub for basic parking stub

- Changed front/src/tests/mountbank/stubs/parking-api.js
  to return a list of parkingSpots as described in the back
  plus their coordinates => ( not yet implemented in back )

fsaintier committed:
Creation cibles specifiques aux actions npm

fsaintier committed:
Update content of README file

fsaintier committed:
Merge branch 'master' of https://github.com/GeoffreyDlvl/Trollop-Project

Commit:
Changed .gitignore for no longer track changes in logs of mountbank

Commit:
Added axios and made call to mountbank for getting list of ParkingSpots

Commit:
Changed stub header in order to allow CORS

Commit:
Add user form (log in & sign in)

Only front elements, back-end needs to be implemented.

Commit:
Add user form (log in & sign in)

Only front elements, back-end needs to be implemented.

Commit:
Implemented display of the points as marker on the map

- Now the app get a list of ParkingSpots from a sources -> API adr
and display them as markers on the map

Commit:
Changed coord to coords in the object returned by the parking API stub

Commit:
Changed coords to have marker in Paris

Commit:
Changed miss positioned latitude and longitude attributes in the array

AdelinoAraujo committed:
test get all data from openData Api

AdelinoAraujo committed:
add OpenDataParis class + getAllParkingSpots method

Commit:
Changed color of the markers

Commit:
Deleted response json from stub

Commit:
Created bigQuery file to teste map with more makers

Commit:
Created a file with the basic response for parkingSpot

Commit:
Try to set up cypress

Commit:
Added e2e spec to check if map and markers appears visible

	- Added test:e2e-headless as headless script on package.json
	- Added tests:e2e-headless as test script to run on travis.yml

Commit:
Deleted useless basic test e2e spec

Commit:
Added e2e screenshots to .gitignore

Commit:
Exported getting all markers in getAllMarkers method

Commit:
Updated travis.yml fixing isssue for npm to run e2e tests

Commit:
Added mountbank start to test script

Commit:
Try to run npm run mountbank-start in background

Commit:
Try to figure out how the run mountbank in travis

Commit:
separated jobs fro travis.yml

Commit:
Changed path

Commit:
npm run mountbank in background

AdelinoAraujo committed:
Add getAllParkingspots

AdelinoAraujo committed:
Add getAllParkingSpotsCars

AdelinoAraujo committed:
Add getAllParkingSpotsMotos

AdelinoAraujo committed:
Add getAllParkingSpotsVelos

AdelinoAraujo committed:
Add getAllParkingSpotsGratuit

AdelinoAraujo committed:
Add getAllParkingSpotsLivraison

AdelinoAraujo committed:
Add getAllParkingSpotsAutocar

AdelinoAraujo committed:
Add getAllParkingSpotsElectrique

AdelinoAraujo committed:
Add filter options for each getAllParkingSpots

Remi-Guijarro committed:
Merge branch 'master' into mapIntegration

Commit:
Fix cypress issues and arrange components map integration into developped boilerplate

Commit:
changed cypress test

AdelinoAraujo committed:
update parkingSpot Class

Commit:
still Change cypress test

AdelinoAraujo committed:
Update parkingSpot.test.js

Commit:
Changed travis.yml

Remi-Guijarro committed:
Merge pull request #21 from GeoffreyDlvl/mapIntegration

Map integration

AdelinoAraujo committed:
Add jsonToObjects for getAllParkingSpots

AdelinoAraujo committed:
Add TODO to ... do

fsaintier committed:
Add library vue-places to autocomplete address text area

fsaintier committed:
Add library vue-places to autocomplete address text area

fsaintier committed:
Inverse the map and the settings and espace them with a spacer

fsaintier committed:
Inverse the map and the settings and espace them with a spacer

fsaintier committed:
Add a search address bar with autocompletion based on vue-places.js

fsaintier committed:
Add a search address bar with autocompletion based on vue-places.js

Commit:
Add axios to project

Axios is a http client library for vue.

Commit:
Add axios to project

Axios is a http client library for vue.

Commit:
Allow CORS to server

Commit:
Allow CORS to server

Commit:
Send user credentials to back end on Log In clicked

Commit:
Send user credentials to back end on Log In clicked

AdelinoAraujo committed:
Add function fromJsontoObject for each openDataPari controller method

AdelinoAraujo committed:
fix error in getAllParkingSpots method

Commit:
Changed card to modal

Commit:
Changed card to modal

Commit:
Merge branch 'userAuthentication' of https://github.com/GeoffreyDlvl/Trollop-Project into userAuthentication

Commit:
Set up nodemon in order to not have to relaunch server every f time

Commit:
Merge branch 'getAllParkingSpots' of https://github.com/GeoffreyDlvl/Trollop-Project into getAllParkingSpots

Commit:
Merge pull request #23 from GeoffreyDlvl/getAllParkingSpots

Get all parking spots

Commit:
Update database architecture

All parking spots won't be saved in database anymore.
Now only reserved parking spots will be written in db.

Commit:
Update database architecture

All parking spots won't be saved in database anymore.
Now only reserved parking spots will be written in db.

Commit:
Merge branch 'mapIntegration' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

Commit:
Changed MakeFile -> start directive in order to not rebuild the app every time -> just start containers

Remi-Guijarro committed:
Merge pull request #24 from GeoffreyDlvl/databaseArchitecture

Update database architecture

fsaintier committed:
Add a shared declaration of the Event-bus to allow front components to exchange datas

fsaintier committed:
Add an Event-bus to allow front components to exchange datas. Not fully operational

fsaintier committed:
Merge branch 'master' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

fsaintier committed:
Merge branch 'mapIntegration' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

Commit:
merge changes and added cors enabler on back

Commit:
Merged changes, last commit was uncomplete merge

fsaintier committed:
Search an address and zoom on it fonctional

fsaintier committed:
Search an address and zoom on it functional
Merge branch 'mapIntegration' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

Commit:
Changed lat and long, markers were in Africa

Commit:
Added new component to display markers informations

Commit:
Implemented communication between MarkerInfo and MapBox in order to display Marker information on click

Commit:
Improved display

- Adress is now capitalized
- Added a badge green if parking spot is available, red badge otherwise
- Fixed postal code display -> if arrond > 9 postal code should be 75 0XX otherwise 75 00X

Commit:
Merge branch 'master' of https://github.com/GeoffreyDlvl/ParkTonChar into userAuthentication

Commit:
Make http base url global in front-end

Commit:
Make http base url global in front-end

Commit:
Changed card layout and style, changed the way component communicate with each other

fsaintier committed:
Decomposition of the onAddressFilled function into 2 functions

Commit:
Modified cypress spec to include wait time and added new cypress spec to test marker information feature

fsaintier committed:
Add handle of empty address search

fsaintier committed:
Merge branch 'mapIntegration' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

fsaintier committed:
Change the center declaration outside of the data function part

fsaintier committed:
Remove handleOnChange useless parameter

Commit:
Set up Https for front app

Commit:
Merge branch 'mapIntegration' of https://github.com/GeoffreyDlvl/Trollop-Project into mapIntegration

AdelinoAraujo committed:
Add supertest t package.json dev-dependencies and get all parking spots test

AdelinoAraujo committed:
getAllParkingSpotsVoitures test

AdelinoAraujo committed:
getAllParkingSpotsMotos test

AdelinoAraujo committed:
getAllParkingSpotsVelos test

AdelinoAraujo committed:
getAllParkingSpotsGratuit test

AdelinoAraujo committed:
getAllParkingSpotsLivraison test

AdelinoAraujo committed:
getAllParkingSpotsAutocar test

AdelinoAraujo committed:
getAllParkingSpotsElectrique test

AdelinoAraujo committed:
Remove getAllParkingSpotsElectrique test to check travis

AdelinoAraujo committed:
ReAdd getAllParkingSpotsElectrique test to check travis

AdelinoAraujo committed:
Merge branch 'master' into updateGetAllParkingSpots

AdelinoAraujo committed:
Add fake test to resolve isue with supertest

Commit:
Added new logo

AdelinoAraujo committed:
Add parameter done to await callback

fsaintier committed:
Correction to allow printing empty addresses

AdelinoAraujo committed:
Add NODE_ENV test condition for port listenneing -www

AdelinoAraujo committed:
Remove _numvoie from getAllParkingSpot test => not always provided

Commit:
Merge pull request #26 from GeoffreyDlvl/updateGetAllParkingSpots

Update get all parking spots

Commit:
Resolve crash on 'make start' command

Commit:
Minor improvements to Makefile and dockerfile

Commit:
Retrieve user from database on connection in home page

AdelinoAraujo committed:
Add caching with ApiCache

Commit:
Merge pull request #27 from GeoffreyDlvl/addApiCache

Add caching with ApiCache

Commit:
Added button Admin

Commit:
Refact : now use of router and added navigation panel to switch between Admin and map vue

Commit:
Added Add user button and table with mocked users

Commit:
Finished boilerplate of admin panel

Commit:
Implement LogIn, LogOut and SignIn

Commit:
Implement authentication controller unit tests

GeoffreyDlvl committed:
Merge branch 'master' into userAuthentication

Commit:
Resolve bug (missing comma) in back package.json

Commit:
Basic user settings

AdelinoAraujo committed:
Update apiCache for openDataParis request only

fsaintier committed:
Add component MapOptions to allow user to filter his search

fsaintier committed:
Merge branch 'master' of https://github.com/GeoffreyDlvl/ParkTonChar into mapIntegration

Commit:
Implemented véhicle to park option for the map

Commit:
Added spinner

Commit:
Implemented filter on sta type

Commit:
Updated spec to pass

Remi-Guijarro committed:
Merge pull request #29 from GeoffreyDlvl/mapIntegration

Map integration

Remi-Guijarro committed:
Merge branch 'master' into userAuthentication

Remi-Guijarro committed:
Merge pull request #28 from GeoffreyDlvl/userAuthentication

User authentication

Commit:
Fixed MapBox sta type option

Commit:
updated package lock

Remi-Guijarro committed:
Merge pull request #30 from GeoffreyDlvl/mapIntegration

Fixed MapBox sta type option

AdelinoAraujo committed:
update local branch with mater + updateApiCache

Commit:
Merge pull request #31 from GeoffreyDlvl/updateApiCache

Update api cache

Commit:
Merge branch 'master' of https://github.com/GeoffreyDlvl/Trollop-Project into adminPanel

rparpa commented on back/src/controller/openDataParisController.js:
```diff
@@ -0,0 +1,189 @@
+
+    var onDataFromOpenDataParisRepo = function (dataFromOpenDataParisRepo){
+
+        electriqueParkingSpotObjectList = fromJsonToObject(dataFromOpenDataParisRepo);
```
Pour info (ptete juste un oubli, mais j'en ai vu plusieurs).

On utilise aujourd'hui deux types de déclarations de variables en js : 
- `let` pour les variables ;
- `const` pour les constantes.

De manière général en js, l'approche immutable est très populaire. On favorise donc l'utilisation de const et de clone d'objets / array.

rparpa commented on back/router.js:
```diff
@@ -0,0 +1,50 @@
+router.post('/updateUser', function(req, res, next) {
+  console.log('REQ BODYYY')
+  console.log(req.body)
+  const controller = new SettingsController(new DatabaseService());
```
Ce n'est pas un controlleur, mais plutôt un service

rparpa commented on back/src/controller/authenticationController.js:
```diff
@@ -0,0 +1,47 @@
+
+    const result = await this.service
+    .read('User', '*', 'username=\''+username+'\' AND password=\''+encryptedPassword+'\'')
+    .then(res => res)
```
Pas besoin de ce then

rparpa commented on back/router.js:
```diff
@@ -0,0 +1,50 @@
+  res.render('index', { title: 'Express' });
+});
+
+router.post('/authentication', function(req, res, next) {
```
Ça ne doit pas vous poser de soucis, car vous n'avez pas de middleware après votre routage mais il est important d'appeler next dans express, surtout pour transmettre une erreur.

La norme par exemple, est de définir un middleware de gestion d'erreur, et de le laisser gérer la réponse / le log etc...

Ça donnerait le code suivant :

```js
(async () => {})().then(stuff => next()).catch(next);
```

rparpa commented on back/src/repository/openDataParisRepository.js:
```diff
@@ -0,0 +1,193 @@
+        });
+    }
+
+    getAllParkingSpotsVoitures(urlOptions,onOpenDataApiReturn){
```
Pour info, on préfère utiliser des promise que des callback dorénavant.

Si vous voulez utiliser un callback en revanche, il faut essayer de suivre le standard JS : le `error first callback`.

Globalement, ça donne une fonction de type suivant :

```js
(error: Error | null, data: TonTypeRetour | null) => void
```

rparpa replied:
ça te permet par exemple de convertir ton callback en promise très facilement avec `util.promisify`.

Commit:
Updated : added State to the app to keep on connection state,
Added redirection when on panel admin and deconnection is triggered

Commit:
Changed data model to include role

Commit:
Add validation sign in form

Commit:
Updated User Entity

Commit:
Updated back routes to include role when user created

Commit:
Updated UserLogVue to implement role on userCreation

Commit:
Disabled https and update for role implementation

Commit:
Displayed the actual list of users in admin panel

Remi-Guijarro committed:
Merge pull request #32 from GeoffreyDlvl/adminPanel

Admin Panel

Commit:
Added button reservation

Commit:
Improve user management

GeoffreyDlvl committed:
Merge branch 'master' into userValidation

Commit:
Resolve minor bug

GeoffreyDlvl committed:
Merge pull request #33 from GeoffreyDlvl/userValidation

User validation

fsaintier committed:
Ajout rapport final
